Failure to Run Buf Action on Github Enterprise

Hi there!

I am trying to use the new Buf Action on Github Enterprise. The action is found and downloaded, but it seems internally, when it tries to install `buf` it uses the `get-latest-release` method to check for the latest Buf release in order to download and install it.
I don't know exactly where the action is looking, but it seems that endpoint does not offer the latest `buf` release and hence the REST API call fails. My guess would be that the endpoint URL is somehow pointing to a Github Enterprise installation, when actually it should point to Github.com - but that's just a guess. 

The error is displayed below. Any ideas / fixes would be much appreciated!


```shell
Run bufbuild/buf-action@v1.0.3
  with:
    input: payload-specifications/audit-logging
    setup_only: true
    domain: buf.build
    github_actor: XXXX
    github_token: ***
    pr_comment: false
    exclude_imports: false
    lint: false
    format: false
    breaking: false
    push: true
    push_disable_create: false
    archive: false
Error: Not Found - https://docs.github.com/enterprise-server@3.12/rest/releases/releases#get-the-latest-release
```

Hey @emcfarlane, thanks, sure!
Here you go:

```
2025-03-07T16:09:11.0721955Z Requested labels: self-hosted, solinas
2025-03-07T16:09:11.0722230Z Job defined at: wg-observability/telemetry-semantic-conventions/.github/workflows/audit-logging-catalog-workflow.yaml@refs/pull/40/merge
2025-03-07T16:09:11.0722326Z Waiting for a runner to pick up this job...
2025-03-07T16:10:03.6773270Z Job is about to start running on the runner: solinas-runner-github-tools-acme-129226-mnkob6 (organization)
2025-03-07T16:10:07.7983863Z Current runner version: '2.322.0'
2025-03-07T16:10:07.7994495Z Runner name: 'solinas-runner-github-tools-acme-129226-mnkob6'
2025-03-07T16:10:07.7996054Z Runner group name: 'sugar-runners'
2025-03-07T16:10:07.7997683Z Machine name: 'gha-runner-github-tools-acme-129226-mnkob6'
2025-03-07T16:10:07.8026001Z ##[group]SUGAR Runner Image
2025-03-07T16:10:07.8027256Z ubuntu2004-20250303114449
2025-03-07T16:10:07.8028089Z ##[endgroup]
2025-03-07T16:10:07.8028828Z ##[group]SUGAR Version
2025-03-07T16:10:07.8030409Z 2.2.0-20250226134316-44703cf87ec4a98feed589dbb4cbdc26e5212541
2025-03-07T16:10:07.8031658Z ##[endgroup]
2025-03-07T16:10:07.8033479Z ##[group]GITHUB_TOKEN Permissions
2025-03-07T16:10:07.8036374Z Contents: read
2025-03-07T16:10:07.8037242Z Metadata: read
2025-03-07T16:10:07.8038150Z PullRequests: write
2025-03-07T16:10:07.8038949Z ##[endgroup]
2025-03-07T16:10:07.8043339Z Secret source: Actions
2025-03-07T16:10:07.8044758Z Prepare workflow directory
2025-03-07T16:10:07.8503912Z Prepare all required actions
2025-03-07T16:10:07.8556351Z Getting action download info
2025-03-07T16:10:09.2199472Z Download action repository 'actions/checkout@v3' (SHA:f43a0e5ff2bd294095638e18286ca9a3d1956744)
2025-03-07T16:10:09.9001897Z Download action repository 'bufbuild/buf-action@v1.0.3' (SHA:b46cbc051dd8fd10a3c2213e784498790d3fe388)
2025-03-07T16:10:10.3590458Z Complete job name: Compile Audit Log Events Catalog
2025-03-07T16:10:10.4222328Z A job started hook has been configured by the self-hosted runner administrator
2025-03-07T16:10:10.4418497Z ##[group]Run '/opt/sugar/job-started.sh'
2025-03-07T16:10:10.4444321Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2025-03-07T16:10:10.4445584Z ##[endgroup]
2025-03-07T16:10:10.4975775Z ##[group]Run actions/checkout@v3
2025-03-07T16:10:10.4976961Z with:
2025-03-07T16:10:10.4977970Z   repository: wg-observability/telemetry-semantic-conventions
2025-03-07T16:10:10.4979433Z   token: ***
2025-03-07T16:10:10.4979938Z   ssh-strict: true
2025-03-07T16:10:10.4980600Z   persist-credentials: true
2025-03-07T16:10:10.4981256Z   clean: true
2025-03-07T16:10:10.4981843Z   sparse-checkout-cone-mode: true
2025-03-07T16:10:10.4982500Z   fetch-depth: 1
2025-03-07T16:10:10.4983057Z   fetch-tags: false
2025-03-07T16:10:10.4983618Z   lfs: false
2025-03-07T16:10:10.4988818Z   submodules: false
2025-03-07T16:10:10.4989385Z   set-safe-directory: true
2025-03-07T16:10:10.4990030Z ##[endgroup]
2025-03-07T16:10:10.6385507Z Syncing repository: wg-observability/telemetry-semantic-conventions
2025-03-07T16:10:10.6388270Z ##[group]Getting Git version info
2025-03-07T16:10:10.6390416Z Working directory is '/opt/actions-runner/_work/telemetry-semantic-conventions/telemetry-semantic-conventions'
2025-03-07T16:10:10.6392124Z [command]/usr/bin/git version
2025-03-07T16:10:10.6514519Z git version 2.25.1
2025-03-07T16:10:10.6559420Z ##[endgroup]
2025-03-07T16:10:10.6581702Z Copying '/home/runner/.gitconfig' to '/opt/actions-runner/_work/_temp/2606f4a5-61c8-4f7b-9219-2adb17abf354/.gitconfig'
2025-03-07T16:10:10.6600644Z Temporarily overriding HOME='/opt/actions-runner/_work/_temp/2606f4a5-61c8-4f7b-9219-2adb17abf354' before making global git config changes
2025-03-07T16:10:10.6602799Z Adding repository directory to the temporary git global config as a safe directory
2025-03-07T16:10:10.6606157Z [command]/usr/bin/git config --global --add safe.directory /opt/actions-runner/_work/telemetry-semantic-conventions/telemetry-semantic-conventions
2025-03-07T16:10:10.6670232Z Deleting the contents of '/opt/actions-runner/_work/telemetry-semantic-conventions/telemetry-semantic-conventions'
2025-03-07T16:10:10.6675269Z ##[group]Initializing the repository
2025-03-07T16:10:10.6679634Z [command]/usr/bin/git init /opt/actions-runner/_work/telemetry-semantic-conventions/telemetry-semantic-conventions
2025-03-07T16:10:10.6827883Z Initialized empty Git repository in /opt/actions-runner/_work/telemetry-semantic-conventions/telemetry-semantic-conventions/.git/
2025-03-07T16:10:10.6842489Z [command]/usr/bin/git remote add origin https://github.tools.acme/wg-observability/telemetry-semantic-conventions
2025-03-07T16:10:10.6891308Z ##[endgroup]
2025-03-07T16:10:10.6892587Z ##[group]Disabling automatic garbage collection
2025-03-07T16:10:10.6895709Z [command]/usr/bin/git config --local gc.auto 0
2025-03-07T16:10:10.6934953Z ##[endgroup]
2025-03-07T16:10:10.6935971Z ##[group]Setting up auth
2025-03-07T16:10:10.6942374Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2025-03-07T16:10:10.6977226Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2025-03-07T16:10:10.7460867Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.tools\.acme\/\.extraheader
2025-03-07T16:10:10.7495868Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.tools\.acme\/\.extraheader' && git config --local --unset-all 'http.https://github.tools.acme/.extraheader' || :"
2025-03-07T16:10:10.8051051Z [command]/usr/bin/git config --local http.https://github.tools.acme/.extraheader AUTHORIZATION: basic ***
2025-03-07T16:10:10.8053888Z ##[endgroup]
2025-03-07T16:10:10.8055062Z ##[group]Fetching the repository
2025-03-07T16:10:10.8057167Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +8c0c8e3548b60c8211f2919f7852bd0e72ce8072:refs/remotes/pull/40/merge
2025-03-07T16:10:11.0477216Z remote: Enumerating objects: 251, done.        
2025-03-07T16:10:11.0478965Z remote: Counting objects:   0% (1/251)        
2025-03-07T16:10:11.0480641Z remote: Counting objects:   1% (3/251)        
2025-03-07T16:10:11.0481772Z remote: Counting objects:   2% (6/251)        
2025-03-07T16:10:11.0482758Z remote: Counting objects:   3% (8/251)        
2025-03-07T16:10:11.0483742Z remote: Counting objects:   4% (11/251)        
2025-03-07T16:10:11.0484734Z remote: Counting objects:   5% (13/251)        
2025-03-07T16:10:11.0485850Z remote: Counting objects:   6% (16/251)        
2025-03-07T16:10:11.0486909Z remote: Counting objects:   7% (18/251)        
2025-03-07T16:10:11.0488126Z remote: Counting objects:   8% (21/251)        
2025-03-07T16:10:11.0489173Z remote: Counting objects:   9% (23/251)        
2025-03-07T16:10:11.0490106Z remote: Counting objects:  10% (26/251)        
2025-03-07T16:10:11.0490725Z remote: Counting objects:  11% (28/251)        
2025-03-07T16:10:11.0491308Z remote: Counting objects:  12% (31/251)        
2025-03-07T16:10:11.0491890Z remote: Counting objects:  13% (33/251)        
2025-03-07T16:10:11.0492502Z remote: Counting objects:  14% (36/251)        
2025-03-07T16:10:11.0493090Z remote: Counting objects:  15% (38/251)        
2025-03-07T16:10:11.0493678Z remote: Counting objects:  16% (41/251)        
2025-03-07T16:10:11.0494249Z remote: Counting objects:  17% (43/251)        
2025-03-07T16:10:11.0494834Z remote: Counting objects:  18% (46/251)        
2025-03-07T16:10:11.0495544Z remote: Counting objects:  19% (48/251)        
2025-03-07T16:10:11.0496462Z remote: Counting objects:  20% (51/251)        
2025-03-07T16:10:11.0497495Z remote: Counting objects:  21% (53/251)        
2025-03-07T16:10:11.0498721Z remote: Counting objects:  22% (56/251)        
2025-03-07T16:10:11.0500046Z remote: Counting objects:  23% (58/251)        
2025-03-07T16:10:11.0501156Z remote: Counting objects:  24% (61/251)        
2025-03-07T16:10:11.0502208Z remote: Counting objects:  25% (63/251)        
2025-03-07T16:10:11.0503245Z remote: Counting objects:  26% (66/251)        
2025-03-07T16:10:11.0504307Z remote: Counting objects:  27% (68/251)        
2025-03-07T16:10:11.0505342Z remote: Counting objects:  28% (71/251)        
2025-03-07T16:10:11.0505936Z remote: Counting objects:  29% (73/251)        
2025-03-07T16:10:11.0506520Z remote: Counting objects:  30% (76/251)        
2025-03-07T16:10:11.0507104Z remote: Counting objects:  31% (78/251)        
2025-03-07T16:10:11.0508123Z remote: Counting objects:  32% (81/251)        
2025-03-07T16:10:11.0508703Z remote: Counting objects:  33% (83/251)        
2025-03-07T16:10:11.0509289Z remote: Counting objects:  34% (86/251)        
2025-03-07T16:10:11.0510096Z remote: Counting objects:  35% (88/251)        
2025-03-07T16:10:11.0511011Z remote: Counting objects:  36% (91/251)        
2025-03-07T16:10:11.0511995Z remote: Counting objects:  37% (93/251)        
2025-03-07T16:10:11.0512979Z remote: Counting objects:  38% (96/251)        
2025-03-07T16:10:11.0513970Z remote: Counting objects:  39% (98/251)        
2025-03-07T16:10:11.0514961Z remote: Counting objects:  40% (101/251)        
2025-03-07T16:10:11.0515848Z remote: Counting objects:  41% (103/251)        
2025-03-07T16:10:11.0516767Z remote: Counting objects:  42% (106/251)        
2025-03-07T16:10:11.0517383Z remote: Counting objects:  43% (108/251)        
2025-03-07T16:10:11.0518098Z remote: Counting objects:  44% (111/251)        
2025-03-07T16:10:11.0518958Z remote: Counting objects:  45% (113/251)        
2025-03-07T16:10:11.0520002Z remote: Counting objects:  46% (116/251)        
2025-03-07T16:10:11.0520953Z remote: Counting objects:  47% (118/251)        
2025-03-07T16:10:11.0521909Z remote: Counting objects:  48% (121/251)        
2025-03-07T16:10:11.0522516Z remote: Counting objects:  49% (123/251)        
2025-03-07T16:10:11.0523210Z remote: Counting objects:  50% (126/251)        
2025-03-07T16:10:11.0524101Z remote: Counting objects:  51% (129/251)        
2025-03-07T16:10:11.0524695Z remote: Counting objects:  52% (131/251)        
2025-03-07T16:10:11.0525365Z remote: Counting objects:  53% (134/251)        
2025-03-07T16:10:11.0526534Z remote: Counting objects:  54% (136/251)        
2025-03-07T16:10:11.0527577Z remote: Counting objects:  55% (139/251)        
2025-03-07T16:10:11.0528631Z remote: Counting objects:  56% (141/251)        
2025-03-07T16:10:11.0529547Z remote: Counting objects:  57% (144/251)        
2025-03-07T16:10:11.0530615Z remote: Counting objects:  58% (146/251)        
2025-03-07T16:10:11.0531369Z remote: Counting objects:  59% (149/251)        
2025-03-07T16:10:11.0531961Z remote: Counting objects:  60% (151/251)        
2025-03-07T16:10:11.0532547Z remote: Counting objects:  61% (154/251)        
2025-03-07T16:10:11.0533131Z remote: Counting objects:  62% (156/251)        
2025-03-07T16:10:11.0533710Z remote: Counting objects:  63% (159/251)        
2025-03-07T16:10:11.0534297Z remote: Counting objects:  64% (161/251)        
2025-03-07T16:10:11.0534890Z remote: Counting objects:  65% (164/251)        
2025-03-07T16:10:11.0535501Z remote: Counting objects:  66% (166/251)        
2025-03-07T16:10:11.0536151Z remote: Counting objects:  67% (169/251)        
2025-03-07T16:10:11.0536733Z remote: Counting objects:  68% (171/251)        
2025-03-07T16:10:11.0537322Z remote: Counting objects:  69% (174/251)        
2025-03-07T16:10:11.0538050Z remote: Counting objects:  70% (176/251)        
2025-03-07T16:10:11.0538639Z remote: Counting objects:  71% (179/251)        
2025-03-07T16:10:11.0541074Z remote: Counting objects:  72% (181/251)        
2025-03-07T16:10:11.0542771Z remote: Counting objects:  73% (184/251)        
2025-03-07T16:10:11.0543781Z remote: Counting objects:  74% (186/251)        
2025-03-07T16:10:11.0544724Z remote: Counting objects:  75% (189/251)        
2025-03-07T16:10:11.0546230Z remote: Counting objects:  76% (191/251)        
2025-03-07T16:10:11.0547355Z remote: Counting objects:  77% (194/251)        
2025-03-07T16:10:11.0548546Z remote: Counting objects:  78% (196/251)        
2025-03-07T16:10:11.0549636Z remote: Counting objects:  79% (199/251)        
2025-03-07T16:10:11.0550709Z remote: Counting objects:  80% (201/251)        
2025-03-07T16:10:11.0551778Z remote: Counting objects:  81% (204/251)        
2025-03-07T16:10:11.0552850Z remote: Counting objects:  82% (206/251)        
2025-03-07T16:10:11.0554039Z remote: Counting objects:  83% (209/251)        
2025-03-07T16:10:11.0555203Z remote: Counting objects:  84% (211/251)        
2025-03-07T16:10:11.0556761Z remote: Counting objects:  85% (214/251)        
2025-03-07T16:10:11.0557789Z remote: Counting objects:  86% (216/251)        
2025-03-07T16:10:11.0559336Z remote: Counting objects:  87% (219/251)        
2025-03-07T16:10:11.0560467Z remote: Counting objects:  88% (221/251)        
2025-03-07T16:10:11.0562375Z remote: Counting objects:  89% (224/251)        
2025-03-07T16:10:11.0563385Z remote: Counting objects:  90% (226/251)        
2025-03-07T16:10:11.0564457Z remote: Counting objects:  91% (229/251)        
2025-03-07T16:10:11.0565056Z remote: Counting objects:  92% (231/251)        
2025-03-07T16:10:11.0565863Z remote: Counting objects:  93% (234/251)        
2025-03-07T16:10:11.0566932Z remote: Counting objects:  94% (236/251)        
2025-03-07T16:10:11.0567930Z remote: Counting objects:  95% (239/251)        
2025-03-07T16:10:11.0569087Z remote: Counting objects:  96% (241/251)        
2025-03-07T16:10:11.0570024Z remote: Counting objects:  97% (244/251)        
2025-03-07T16:10:11.0570996Z remote: Counting objects:  98% (246/251)        
2025-03-07T16:10:11.0572028Z remote: Counting objects:  99% (249/251)        
2025-03-07T16:10:11.0573140Z remote: Counting objects: 100% (251/251)        
2025-03-07T16:10:11.0574062Z remote: Counting objects: 100% (251/251), done.        
2025-03-07T16:10:11.0575104Z remote: Compressing objects:   0% (1/200)        
2025-03-07T16:10:11.0576169Z remote: Compressing objects:   1% (2/200)        
2025-03-07T16:10:11.0576958Z remote: Compressing objects:   2% (4/200)        
2025-03-07T16:10:11.0578051Z remote: Compressing objects:   3% (6/200)        
2025-03-07T16:10:11.0579622Z remote: Compressing objects:   4% (8/200)        
2025-03-07T16:10:11.0580487Z remote: Compressing objects:   5% (10/200)        
2025-03-07T16:10:11.0581609Z remote: Compressing objects:   6% (12/200)        
2025-03-07T16:10:11.0582613Z remote: Compressing objects:   7% (14/200)        
2025-03-07T16:10:11.0583659Z remote: Compressing objects:   8% (16/200)        
2025-03-07T16:10:11.0584701Z remote: Compressing objects:   9% (18/200)        
2025-03-07T16:10:11.0585833Z remote: Compressing objects:  10% (20/200)        
2025-03-07T16:10:11.0586569Z remote: Compressing objects:  11% (22/200)        
2025-03-07T16:10:11.0587229Z remote: Compressing objects:  12% (24/200)        
2025-03-07T16:10:11.0587934Z remote: Compressing objects:  13% (26/200)        
2025-03-07T16:10:11.0588649Z remote: Compressing objects:  14% (28/200)        
2025-03-07T16:10:11.0589296Z remote: Compressing objects:  15% (30/200)        
2025-03-07T16:10:11.0589918Z remote: Compressing objects:  16% (32/200)        
2025-03-07T16:10:11.0590785Z remote: Compressing objects:  17% (34/200)        
2025-03-07T16:10:11.0591400Z remote: Compressing objects:  18% (36/200)        
2025-03-07T16:10:11.0592012Z remote: Compressing objects:  19% (38/200)        
2025-03-07T16:10:11.0592967Z remote: Compressing objects:  20% (40/200)        
2025-03-07T16:10:11.0593952Z remote: Compressing objects:  21% (42/200)        
2025-03-07T16:10:11.0595079Z remote: Compressing objects:  22% (44/200)        
2025-03-07T16:10:11.0596097Z remote: Compressing objects:  23% (46/200)        
2025-03-07T16:10:11.0597202Z remote: Compressing objects:  24% (48/200)        
2025-03-07T16:10:11.0598312Z remote: Compressing objects:  25% (50/200)        
2025-03-07T16:10:11.0599221Z remote: Compressing objects:  26% (52/200)        
2025-03-07T16:10:11.0600090Z remote: Compressing objects:  27% (54/200)        
2025-03-07T16:10:11.0601171Z remote: Compressing objects:  28% (56/200)        
2025-03-07T16:10:11.0602138Z remote: Compressing objects:  29% (58/200)        
2025-03-07T16:10:11.0602948Z remote: Compressing objects:  30% (60/200)        
2025-03-07T16:10:11.0603569Z remote: Compressing objects:  31% (62/200)        
2025-03-07T16:10:11.0604183Z remote: Compressing objects:  32% (64/200)        
2025-03-07T16:10:11.0604790Z remote: Compressing objects:  33% (66/200)        
2025-03-07T16:10:11.0605389Z remote: Compressing objects:  34% (68/200)        
2025-03-07T16:10:11.0606319Z remote: Compressing objects:  35% (70/200)        
2025-03-07T16:10:11.0606928Z remote: Compressing objects:  36% (72/200)        
2025-03-07T16:10:11.0607530Z remote: Compressing objects:  37% (74/200)        
2025-03-07T16:10:11.0609010Z remote: Compressing objects:  38% (76/200)        
2025-03-07T16:10:11.0610185Z remote: Compressing objects:  39% (78/200)        
2025-03-07T16:10:11.0611167Z remote: Compressing objects:  40% (80/200)        
2025-03-07T16:10:11.0612283Z remote: Compressing objects:  41% (82/200)        
2025-03-07T16:10:11.0613400Z remote: Compressing objects:  42% (84/200)        
2025-03-07T16:10:11.0614545Z remote: Compressing objects:  43% (86/200)        
2025-03-07T16:10:11.0615657Z remote: Compressing objects:  44% (88/200)        
2025-03-07T16:10:11.0616770Z remote: Compressing objects:  45% (90/200)        
2025-03-07T16:10:11.0617852Z remote: Compressing objects:  46% (92/200)        
2025-03-07T16:10:11.0618997Z remote: Compressing objects:  47% (94/200)        
2025-03-07T16:10:11.0620295Z remote: Compressing objects:  48% (96/200)        
2025-03-07T16:10:11.0621029Z remote: Compressing objects:  49% (98/200)        
2025-03-07T16:10:11.0622094Z remote: Compressing objects:  50% (100/200)        
2025-03-07T16:10:11.0622801Z remote: Compressing objects:  51% (102/200)        
2025-03-07T16:10:11.0623583Z remote: Compressing objects:  52% (104/200)        
2025-03-07T16:10:11.0624688Z remote: Compressing objects:  53% (106/200)        
2025-03-07T16:10:11.0625357Z remote: Compressing objects:  54% (108/200)        
2025-03-07T16:10:11.0626814Z remote: Compressing objects:  55% (110/200)        
2025-03-07T16:10:11.0627753Z remote: Compressing objects:  56% (112/200)        
2025-03-07T16:10:11.0628491Z remote: Compressing objects:  57% (114/200)        
2025-03-07T16:10:11.0629411Z remote: Compressing objects:  58% (116/200)        
2025-03-07T16:10:11.0630509Z remote: Compressing objects:  59% (118/200)        
2025-03-07T16:10:11.0631252Z remote: Compressing objects:  60% (120/200)        
2025-03-07T16:10:11.0632240Z remote: Compressing objects:  61% (122/200)        
2025-03-07T16:10:11.0633303Z remote: Compressing objects:  62% (124/200)        
2025-03-07T16:10:11.0634416Z remote: Compressing objects:  63% (126/200)        
2025-03-07T16:10:11.0635265Z remote: Compressing objects:  64% (128/200)        
2025-03-07T16:10:11.0636225Z remote: Compressing objects:  65% (130/200)        
2025-03-07T16:10:11.0637328Z remote: Compressing objects:  66% (132/200)        
2025-03-07T16:10:11.0638370Z remote: Compressing objects:  67% (134/200)        
2025-03-07T16:10:11.0639482Z remote: Compressing objects:  68% (136/200)        
2025-03-07T16:10:11.0640189Z remote: Compressing objects:  69% (138/200)        
2025-03-07T16:10:11.0641262Z remote: Compressing objects:  70% (140/200)        
2025-03-07T16:10:11.0642099Z remote: Compressing objects:  71% (142/200)        
2025-03-07T16:10:11.0642712Z remote: Compressing objects:  72% (144/200)        
2025-03-07T16:10:11.0643407Z remote: Compressing objects:  73% (146/200)        
2025-03-07T16:10:11.0644545Z remote: Compressing objects:  74% (148/200)        
2025-03-07T16:10:11.0645225Z remote: Compressing objects:  75% (150/200)        
2025-03-07T16:10:11.0646278Z remote: Compressing objects:  76% (152/200)        
2025-03-07T16:10:11.0647406Z remote: Compressing objects:  77% (154/200)        
2025-03-07T16:10:11.0648320Z remote: Compressing objects:  78% (156/200)        
2025-03-07T16:10:11.0649336Z remote: Compressing objects:  79% (158/200)        
2025-03-07T16:10:11.0649985Z remote: Compressing objects:  80% (160/200)        
2025-03-07T16:10:11.0651093Z remote: Compressing objects:  81% (162/200)        
2025-03-07T16:10:11.0652147Z remote: Compressing objects:  82% (164/200)        
2025-03-07T16:10:11.0653208Z remote: Compressing objects:  83% (166/200)        
2025-03-07T16:10:11.0654182Z remote: Compressing objects:  84% (168/200)        
2025-03-07T16:10:11.0655701Z remote: Compressing objects:  85% (170/200)        
2025-03-07T16:10:11.0656625Z remote: Compressing objects:  86% (172/200)        
2025-03-07T16:10:11.0657712Z remote: Compressing objects:  87% (174/200)        
2025-03-07T16:10:11.0658683Z remote: Compressing objects:  88% (176/200)        
2025-03-07T16:10:11.0659610Z remote: Compressing objects:  89% (178/200)        
2025-03-07T16:10:11.0660441Z remote: Compressing objects:  90% (180/200)        
2025-03-07T16:10:11.0661349Z remote: Compressing objects:  91% (182/200)        
2025-03-07T16:10:11.0662238Z remote: Compressing objects:  92% (184/200)        
2025-03-07T16:10:11.0663253Z remote: Compressing objects:  93% (186/200)        
2025-03-07T16:10:11.0664183Z remote: Compressing objects:  94% (188/200)        
2025-03-07T16:10:11.0665140Z remote: Compressing objects:  95% (190/200)        
2025-03-07T16:10:11.0666146Z remote: Compressing objects:  96% (192/200)        
2025-03-07T16:10:11.0667082Z remote: Compressing objects:  97% (194/200)        
2025-03-07T16:10:11.0668026Z remote: Compressing objects:  98% (196/200)        
2025-03-07T16:10:11.0669020Z remote: Compressing objects:  99% (198/200)        
2025-03-07T16:10:11.0669869Z remote: Compressing objects: 100% (200/200)        
2025-03-07T16:10:11.0670927Z remote: Compressing objects: 100% (200/200), done.        
2025-03-07T16:10:11.0671832Z Receiving objects:   0% (1/251)
2025-03-07T16:10:11.0672709Z Receiving objects:   1% (3/251)
2025-03-07T16:10:11.0673478Z Receiving objects:   2% (6/251)
2025-03-07T16:10:11.0674223Z Receiving objects:   3% (8/251)
2025-03-07T16:10:11.0674931Z Receiving objects:   4% (11/251)
2025-03-07T16:10:11.0676060Z Receiving objects:   5% (13/251)
2025-03-07T16:10:11.0676984Z Receiving objects:   6% (16/251)
2025-03-07T16:10:11.0677763Z Receiving objects:   7% (18/251)
2025-03-07T16:10:11.0679524Z Receiving objects:   8% (21/251)
2025-03-07T16:10:11.0680510Z Receiving objects:   9% (23/251)
2025-03-07T16:10:11.0681742Z Receiving objects:  10% (26/251)
2025-03-07T16:10:11.0682645Z Receiving objects:  11% (28/251)
2025-03-07T16:10:11.0683481Z Receiving objects:  12% (31/251)
2025-03-07T16:10:11.0690765Z Receiving objects:  13% (33/251)
2025-03-07T16:10:11.0692583Z Receiving objects:  14% (36/251)
2025-03-07T16:10:11.0699709Z Receiving objects:  15% (38/251)
2025-03-07T16:10:11.0701015Z Receiving objects:  16% (41/251)
2025-03-07T16:10:11.0702286Z Receiving objects:  17% (43/251)
2025-03-07T16:10:11.0703194Z Receiving objects:  18% (46/251)
2025-03-07T16:10:11.0705861Z Receiving objects:  19% (48/251)
2025-03-07T16:10:11.0706685Z Receiving objects:  20% (51/251)
2025-03-07T16:10:11.0708001Z Receiving objects:  21% (53/251)
2025-03-07T16:10:11.0709446Z Receiving objects:  22% (56/251)
2025-03-07T16:10:11.0710773Z Receiving objects:  23% (58/251)
2025-03-07T16:10:11.0716322Z Receiving objects:  24% (61/251)
2025-03-07T16:10:11.0718656Z Receiving objects:  25% (63/251)
2025-03-07T16:10:11.0719814Z Receiving objects:  26% (66/251)
2025-03-07T16:10:11.0721343Z Receiving objects:  27% (68/251)
2025-03-07T16:10:11.0723636Z Receiving objects:  28% (71/251)
2025-03-07T16:10:11.0727384Z Receiving objects:  29% (73/251)
2025-03-07T16:10:11.0733550Z Receiving objects:  30% (76/251)
2025-03-07T16:10:11.0735148Z Receiving objects:  31% (78/251)
2025-03-07T16:10:11.0736690Z Receiving objects:  32% (81/251)
2025-03-07T16:10:11.0739436Z Receiving objects:  33% (83/251)
2025-03-07T16:10:11.0747590Z Receiving objects:  34% (86/251)
2025-03-07T16:10:11.0748436Z Receiving objects:  35% (88/251)
2025-03-07T16:10:11.0749224Z Receiving objects:  36% (91/251)
2025-03-07T16:10:11.0752269Z Receiving objects:  37% (93/251)
2025-03-07T16:10:11.0812973Z Receiving objects:  38% (96/251)
2025-03-07T16:10:11.0874774Z Receiving objects:  39% (98/251)
2025-03-07T16:10:11.0918750Z Receiving objects:  40% (101/251)
2025-03-07T16:10:11.0919785Z Receiving objects:  41% (103/251)
2025-03-07T16:10:11.0920911Z Receiving objects:  42% (106/251)
2025-03-07T16:10:11.0964267Z Receiving objects:  43% (108/251)
2025-03-07T16:10:11.0981833Z Receiving objects:  44% (111/251)
2025-03-07T16:10:11.0982832Z Receiving objects:  45% (113/251)
2025-03-07T16:10:11.0985623Z Receiving objects:  46% (116/251)
2025-03-07T16:10:11.0986534Z Receiving objects:  47% (118/251)
2025-03-07T16:10:11.0987461Z Receiving objects:  48% (121/251)
2025-03-07T16:10:11.0988451Z Receiving objects:  49% (123/251)
2025-03-07T16:10:11.0997436Z Receiving objects:  50% (126/251)
2025-03-07T16:10:11.0998480Z Receiving objects:  51% (129/251)
2025-03-07T16:10:11.0999362Z Receiving objects:  52% (131/251)
2025-03-07T16:10:11.1000218Z Receiving objects:  53% (134/251)
2025-03-07T16:10:11.1002973Z Receiving objects:  54% (136/251)
2025-03-07T16:10:11.1003925Z Receiving objects:  55% (139/251)
2025-03-07T16:10:11.1004769Z Receiving objects:  56% (141/251)
2025-03-07T16:10:11.1005622Z Receiving objects:  57% (144/251)
2025-03-07T16:10:11.1006511Z Receiving objects:  58% (146/251)
2025-03-07T16:10:11.1007338Z Receiving objects:  59% (149/251)
2025-03-07T16:10:11.1008221Z Receiving objects:  60% (151/251)
2025-03-07T16:10:11.1009057Z Receiving objects:  61% (154/251)
2025-03-07T16:10:11.1016754Z Receiving objects:  62% (156/251)
2025-03-07T16:10:11.1017467Z Receiving objects:  63% (159/251)
2025-03-07T16:10:11.1018424Z Receiving objects:  64% (161/251)
2025-03-07T16:10:11.1022621Z Receiving objects:  65% (164/251)
2025-03-07T16:10:11.1023919Z remote: Total 251 (delta 44), reused 173 (delta 36), pack-reused 0        
2025-03-07T16:10:11.1027256Z Receiving objects:  66% (166/251)
2025-03-07T16:10:11.1029045Z Receiving objects:  67% (169/251)
2025-03-07T16:10:11.1029922Z Receiving objects:  68% (171/251)
2025-03-07T16:10:11.1031150Z Receiving objects:  69% (174/251)
2025-03-07T16:10:11.1032024Z Receiving objects:  70% (176/251)
2025-03-07T16:10:11.1032899Z Receiving objects:  71% (179/251)
2025-03-07T16:10:11.1035453Z Receiving objects:  72% (181/251)
2025-03-07T16:10:11.1036353Z Receiving objects:  73% (184/251)
2025-03-07T16:10:11.1037219Z Receiving objects:  74% (186/251)
2025-03-07T16:10:11.1039680Z Receiving objects:  75% (189/251)
2025-03-07T16:10:11.1043731Z Receiving objects:  76% (191/251)
2025-03-07T16:10:11.1044895Z Receiving objects:  77% (194/251)
2025-03-07T16:10:11.1046252Z Receiving objects:  78% (196/251)
2025-03-07T16:10:11.1047484Z Receiving objects:  79% (199/251)
2025-03-07T16:10:11.1049411Z Receiving objects:  80% (201/251)
2025-03-07T16:10:11.1050494Z Receiving objects:  81% (204/251)
2025-03-07T16:10:11.1051568Z Receiving objects:  82% (206/251)
2025-03-07T16:10:11.1052648Z Receiving objects:  83% (209/251)
2025-03-07T16:10:11.1053785Z Receiving objects:  84% (211/251)
2025-03-07T16:10:11.1054800Z Receiving objects:  85% (214/251)
2025-03-07T16:10:11.1056004Z Receiving objects:  86% (216/251)
2025-03-07T16:10:11.1057058Z Receiving objects:  87% (219/251)
2025-03-07T16:10:11.1058217Z Receiving objects:  88% (221/251)
2025-03-07T16:10:11.1059508Z Receiving objects:  89% (224/251)
2025-03-07T16:10:11.1060567Z Receiving objects:  90% (226/251)
2025-03-07T16:10:11.1061734Z Receiving objects:  91% (229/251)
2025-03-07T16:10:11.1062774Z Receiving objects:  92% (231/251)
2025-03-07T16:10:11.1063734Z Receiving objects:  93% (234/251)
2025-03-07T16:10:11.1064747Z Receiving objects:  94% (236/251)
2025-03-07T16:10:11.1065546Z Receiving objects:  95% (239/251)
2025-03-07T16:10:11.1066339Z Receiving objects:  96% (241/251)
2025-03-07T16:10:11.1067342Z Receiving objects:  97% (244/251)
2025-03-07T16:10:11.1068455Z Receiving objects:  98% (246/251)
2025-03-07T16:10:11.1069230Z Receiving objects:  99% (249/251)
2025-03-07T16:10:11.1069724Z Receiving objects: 100% (251/251)
2025-03-07T16:10:11.1070395Z Receiving objects: 100% (251/251), 1.62 MiB | 30.15 MiB/s, done.
2025-03-07T16:10:11.1071082Z Resolving deltas:   0% (0/44)
2025-03-07T16:10:11.1071555Z Resolving deltas:   2% (1/44)
2025-03-07T16:10:11.1072022Z Resolving deltas:   4% (2/44)
2025-03-07T16:10:11.1081430Z Resolving deltas:   6% (3/44)
2025-03-07T16:10:11.1082310Z Resolving deltas:  18% (8/44)
2025-03-07T16:10:11.1083595Z Resolving deltas:  20% (9/44)
2025-03-07T16:10:11.1084354Z Resolving deltas:  25% (11/44)
2025-03-07T16:10:11.1085154Z Resolving deltas:  43% (19/44)
2025-03-07T16:10:11.1085871Z Resolving deltas:  47% (21/44)
2025-03-07T16:10:11.1086649Z Resolving deltas:  54% (24/44)
2025-03-07T16:10:11.1087463Z Resolving deltas:  56% (25/44)
2025-03-07T16:10:11.1088299Z Resolving deltas:  59% (26/44)
2025-03-07T16:10:11.1089058Z Resolving deltas:  61% (27/44)
2025-03-07T16:10:11.1089882Z Resolving deltas:  63% (28/44)
2025-03-07T16:10:11.1090619Z Resolving deltas:  70% (31/44)
2025-03-07T16:10:11.1091342Z Resolving deltas:  72% (32/44)
2025-03-07T16:10:11.1092128Z Resolving deltas:  75% (33/44)
2025-03-07T16:10:11.1092878Z Resolving deltas:  77% (34/44)
2025-03-07T16:10:11.1093622Z Resolving deltas:  79% (35/44)
2025-03-07T16:10:11.1094386Z Resolving deltas:  81% (36/44)
2025-03-07T16:10:11.1095153Z Resolving deltas:  84% (37/44)
2025-03-07T16:10:11.1095952Z Resolving deltas:  86% (38/44)
2025-03-07T16:10:11.1096470Z Resolving deltas:  88% (39/44)
2025-03-07T16:10:11.1096922Z Resolving deltas:  90% (40/44)
2025-03-07T16:10:11.1097379Z Resolving deltas:  93% (41/44)
2025-03-07T16:10:11.1097857Z Resolving deltas:  95% (42/44)
2025-03-07T16:10:11.1102240Z Resolving deltas:  97% (43/44)
2025-03-07T16:10:11.1103092Z Resolving deltas: 100% (44/44)
2025-03-07T16:10:11.1103951Z Resolving deltas: 100% (44/44), done.
2025-03-07T16:10:11.1267809Z From https://github.tools.acme/wg-observability/telemetry-semantic-conventions
2025-03-07T16:10:11.1270261Z  * [new ref]         8c0c8e3548b60c8211f2919f7852bd0e72ce8072 -> pull/40/merge
2025-03-07T16:10:11.1300064Z ##[endgroup]
2025-03-07T16:10:11.1301165Z ##[group]Determining the checkout info
2025-03-07T16:10:11.1302318Z ##[endgroup]
2025-03-07T16:10:11.1303503Z ##[group]Checking out the ref
2025-03-07T16:10:11.1307478Z [command]/usr/bin/git checkout --progress --force refs/remotes/pull/40/merge
2025-03-07T16:10:11.1719359Z Note: switching to 'refs/remotes/pull/40/merge'.
2025-03-07T16:10:11.1720784Z 
2025-03-07T16:10:11.1721623Z You are in 'detached HEAD' state. You can look around, make experimental
2025-03-07T16:10:11.1723145Z changes and commit them, and you can discard any commits you make in this
2025-03-07T16:10:11.1724020Z state without impacting any branches by switching back to a branch.
2025-03-07T16:10:11.1724539Z 
2025-03-07T16:10:11.1724905Z If you want to create a new branch to retain commits you create, you may
2025-03-07T16:10:11.1725704Z do so (now or later) by using -c with the switch command. Example:
2025-03-07T16:10:11.1726156Z 
2025-03-07T16:10:11.1726494Z   git switch -c <new-branch-name>
2025-03-07T16:10:11.1727019Z 
2025-03-07T16:10:11.1727389Z Or undo this operation with:
2025-03-07T16:10:11.1727865Z 
2025-03-07T16:10:11.1728073Z   git switch -
2025-03-07T16:10:11.1728297Z 
2025-03-07T16:10:11.1728695Z Turn off this advice by setting config variable advice.detachedHead to false
2025-03-07T16:10:11.1729250Z 
2025-03-07T16:10:11.1729946Z HEAD is now at 8c0c8e3 Merge 7928208a49cfcf3d86b84a1d3187f807a2b595d2 into 1e62e7dcbdf31e69c0adae3c445f5a969aceb1ba
2025-03-07T16:10:11.1732121Z ##[endgroup]
2025-03-07T16:10:11.1789068Z [command]/usr/bin/git log -1 --format='%H'
2025-03-07T16:10:11.1818226Z '8c0c8e3548b60c8211f2919f7852bd0e72ce8072'
2025-03-07T16:10:11.2145337Z ##[group]Run bufbuild/buf-action@v1.0.3
2025-03-07T16:10:11.2146316Z with:
2025-03-07T16:10:11.2147038Z   input: payload-specifications/audit-logging
2025-03-07T16:10:11.2147869Z   setup_only: true
2025-03-07T16:10:11.2148258Z   domain: buf.build
2025-03-07T16:10:11.2148646Z   github_actor: xxxxx
2025-03-07T16:10:11.2149526Z   github_token: ***
2025-03-07T16:10:11.2149926Z   pr_comment: true
2025-03-07T16:10:11.2150358Z   exclude_imports: false
2025-03-07T16:10:11.2150768Z   lint: true
2025-03-07T16:10:11.2151119Z   format: true
2025-03-07T16:10:11.2151469Z   breaking: true
2025-03-07T16:10:11.2151846Z   push: false
2025-03-07T16:10:11.2152228Z   push_disable_create: false
2025-03-07T16:10:11.2152657Z   archive: false
2025-03-07T16:10:11.2153371Z ##[endgroup]
2025-03-07T16:10:11.6009728Z ##[error]Not Found - https://docs.github.com/enterprise-server@3.12/rest/releases/releases#get-the-latest-release
2025-03-07T16:10:11.7081703Z Post job cleanup.
2025-03-07T16:10:11.8135101Z [command]/usr/bin/git version
2025-03-07T16:10:11.8183115Z git version 2.25.1
2025-03-07T16:10:11.8237542Z Copying '/home/runner/.gitconfig' to '/opt/actions-runner/_work/_temp/950db608-3c3e-4f81-9bdb-4e2e4e4749f7/.gitconfig'
2025-03-07T16:10:11.8253680Z Temporarily overriding HOME='/opt/actions-runner/_work/_temp/950db608-3c3e-4f81-9bdb-4e2e4e4749f7' before making global git config changes
2025-03-07T16:10:11.8255625Z Adding repository directory to the temporary git global config as a safe directory
2025-03-07T16:10:11.8260899Z [command]/usr/bin/git config --global --add safe.directory /opt/actions-runner/_work/telemetry-semantic-conventions/telemetry-semantic-conventions
2025-03-07T16:10:11.8305847Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2025-03-07T16:10:11.8342028Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2025-03-07T16:10:11.8653577Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.tools\.acme\/\.extraheader
2025-03-07T16:10:11.8679576Z http.https://github.tools.acme/.extraheader
2025-03-07T16:10:11.8694907Z [command]/usr/bin/git config --local --unset-all http.https://github.tools.acme/.extraheader
2025-03-07T16:10:11.8734137Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.tools\.acme\/\.extraheader' && git config --local --unset-all 'http.https://github.tools.acme/.extraheader' || :"
2025-03-07T16:10:11.9199007Z Cleaning up orphan processes
```

Your action is allowed by Github enterprise, it can also be successfully downloaded. It is really an issue of executing it.

--- a/action.yml
+++ b/action.yml
@@ -50,7 +50,7 @@ inputs:
     default: ${{ github.token }}
   public_github_token:
     description: |-
-      GitHub token for github.com. Required to be set when running on a private GitHub Enterprise instance, and ignored otherwise.
+      GitHub token for github.com. Must be set when running on a private GitHub Enterprise instance to authenticate requests, otherwise ignored.
   setup_only:
     description: |-
       Setup only the buf environment, optionally logging into the BSR, but without executing other commands.

--- a/action.yml
+++ b/action.yml
@@ -45,12 +45,12 @@ inputs:
     default: ${{ github.actor }}
   github_token:
     description: |-
-      GitHub token for API requests.
+      GitHub token for the GitHub instance this action is running on.
     required: false
     default: ${{ github.token }}
   public_github_token:
     description: |-
-      GitHub token for the public github.com, for use with enterprise GitHub instances.
+      GitHub token for github.com. Required to be set when running on a private GitHub Enterprise instance, and ignored otherwise.
   setup_only:
     description: |-
       Setup only the buf environment, optionally logging into the BSR, but without executing other commands.

--- a/action.yml
+++ b/action.yml
@@ -48,6 +48,9 @@ inputs:
       GitHub token for API requests.
     required: false
     default: ${{ github.token }}
+  public_github_token:
+    description: |-
+      GitHub token for the public github.com, for use with enterprise GitHub instances.
   setup_only:
     description: |-
       Setup only the buf environment, optionally logging into the BSR, but without executing other commands.

--- a/src/inputs.ts
+++ b/src/inputs.ts
@@ -29,6 +29,7 @@ export interface Inputs {
   pr_comment: boolean;
   github_actor: string;
   github_token: string;
+  public_github_token: string;
 
   input: string;
   paths: string[];
@@ -55,6 +56,7 @@ export function getInputs(): Inputs {
     pr_comment: core.getBooleanInput("pr_comment"),
     github_actor: core.getInput("github_actor"),
     github_token: core.getInput("github_token"),
+    public_github_token: core.getInput("public_github_token"),
     // Inputs shared between buf steps.
     input: core.getInput("input"),
     paths: core.getMultilineInput("paths"),

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -15,10 +15,9 @@
 import * as core from "@actions/core";
 import * as exec from "@actions/exec";
 import * as tc from "@actions/tool-cache";
-import { getOctokit } from "@actions/github";
 import { GitHub } from "@actions/github/lib/utils";
 import * as semver from "semver";
-import { getEnv, Inputs } from "./inputs";
+import { getEnv } from "./inputs";
 
 // requiredVersion is the minimum version of buf required.
 const requiredVersion = ">=1.35.0";
@@ -68,7 +67,7 @@ export async function installBuf(
   let cachePath = tc.find(bufName, resolvedVersion);
   if (!cachePath) {
     core.info(`Downloading buf (${resolvedVersion})`);
-    const downloadPath = await downloadBuf(resolvedVersion, publicGitHubToken);
+    const downloadPath = await downloadBuf(resolvedVersion, githubToken);
     await exec.exec("chmod", ["+x", downloadPath]);
     cachePath = await tc.cacheFile(
       downloadPath,
@@ -120,7 +119,7 @@ type platformTable = {
 };
 
 // downloadBuf downloads the buf binary and returns the path to the binary.
-async function downloadBuf(version: string, github_token: string): Promise<string> {
+async function downloadBuf(version: string, githubToken: string): Promise<string> {
   const table: platformTable = {
     darwin: {
       x64: "buf-Darwin-x86_64",
@@ -148,8 +147,9 @@ async function downloadBuf(version: string, github_token: string): Promise<strin
     );
   }
   const downloadURL = `https://github.com/bufbuild/buf/releases/download/v${version}/${executable}`;
+  const auth = githubToken ? `token ${githubToken}` : undefined;
   try {
-    return await tc.downloadTool(downloadURL, undefined, github_token);
+    return await tc.downloadTool(downloadURL, undefined, auth);
   } catch (error) {
     throw new Error(
       `Failed to download buf version ${version} from "${downloadURL}": ${error}`,

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -15,20 +15,24 @@
 import * as core from "@actions/core";
 import * as exec from "@actions/exec";
 import * as tc from "@actions/tool-cache";
+import { getOctokit } from "@actions/github";
 import { GitHub } from "@actions/github/lib/utils";
 import * as semver from "semver";
-import { getEnv } from "./inputs";
+import { getEnv, Inputs } from "./inputs";
 
 // requiredVersion is the minimum version of buf required.
 const requiredVersion = ">=1.35.0";
+
+// URL for the public GitHub API.
+const publicGitHubApiUrl = "https://api.github.com";
 
 // installBuf installs the buf binary and returns the path to the binary. The
 // versionInput should be an explicit version of buf.
 export async function installBuf(
   github: InstanceType<typeof GitHub>,
-  versionInput: string,
+  inputs: Inputs,
 ): Promise<[string, string]> {
-  let resolvedVersion = resolveVersion(versionInput);
+  let resolvedVersion = resolveVersion(inputs.version);
   if (resolvedVersion != "" && !tc.isExplicitVersion(resolvedVersion)) {
     throw new Error(
       `The version provided must be an explicit version: ${resolvedVersion}`,
@@ -55,7 +59,8 @@ export async function installBuf(
     }
   }
   if (resolvedVersion === "") {
-    resolvedVersion = await latestVersion(github);
+    const publicGitHub = resolvePublicGitHub(github, inputs);
+    resolvedVersion = await latestVersion(publicGitHub);
   }
   if (!semver.satisfies(resolvedVersion, requiredVersion)) {
     throw new Error(
@@ -78,6 +83,26 @@ export async function installBuf(
   core.addPath(cachePath);
   core.info(`Setup buf (${resolvedVersion}) at ${cachePath}`);
   return [binName, resolvedVersion];
+}
+
+// resolvePublicGitHub returns a GitHub instance for the public GitHub API if
+// the environment is a GitHub Enterprise instance.
+function resolvePublicGitHub(
+  github: InstanceType<typeof GitHub>,
+  inputs: Inputs,
+): InstanceType<typeof GitHub> {
+  const apiUrl = process.env.GITHUB_API_URL || ``;
+  if (apiUrl.startsWith(publicGitHubApiUrl)) {
+    core.info("Running on GitHub.com, using default GitHub API.");
+    return github;
+  }
+  const publicToken = inputs.public_github_token;
+  if (!publicToken) {
+    throw new Error("public_github_token is required for public GitHub API");
+  }
+  const publicGitHub = getOctokit(publicToken, { baseUrl: publicGitHubApiUrl });
+  core.info("Running on GitHub Enterprise, using public GitHub API.");
+  return publicGitHub;
 }
 
 // resolveVersion from the input or environment.

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -23,16 +23,14 @@ import { getEnv, Inputs } from "./inputs";
 // requiredVersion is the minimum version of buf required.
 const requiredVersion = ">=1.35.0";
 
-// URL for the public GitHub API.
-const publicGitHubApiUrl = "https://api.github.com";
-
 // installBuf installs the buf binary and returns the path to the binary. The
 // versionInput should be an explicit version of buf.
 export async function installBuf(
   github: InstanceType<typeof GitHub>,
-  inputs: Inputs,
+  githubToken: string,
+  inputVersion: string,
 ): Promise<[string, string]> {
-  let resolvedVersion = resolveVersion(inputs.version);
+  let resolvedVersion = resolveVersion(inputVersion);
   if (resolvedVersion != "" && !tc.isExplicitVersion(resolvedVersion)) {
     throw new Error(
       `The version provided must be an explicit version: ${resolvedVersion}`,
@@ -59,8 +57,7 @@ export async function installBuf(
     }
   }
   if (resolvedVersion === "") {
-    const publicGitHub = resolvePublicGitHub(github, inputs);
-    resolvedVersion = await latestVersion(publicGitHub);
+    resolvedVersion = await latestVersion(github);
   }
   if (!semver.satisfies(resolvedVersion, requiredVersion)) {
     throw new Error(
@@ -71,7 +68,7 @@ export async function installBuf(
   let cachePath = tc.find(bufName, resolvedVersion);
   if (!cachePath) {
     core.info(`Downloading buf (${resolvedVersion})`);
-    const downloadPath = await downloadBuf(resolvedVersion);
+    const downloadPath = await downloadBuf(resolvedVersion, publicGitHubToken);
     await exec.exec("chmod", ["+x", downloadPath]);
     cachePath = await tc.cacheFile(
       downloadPath,
@@ -83,26 +80,6 @@ export async function installBuf(
   core.addPath(cachePath);
   core.info(`Setup buf (${resolvedVersion}) at ${cachePath}`);
   return [binName, resolvedVersion];
-}
-
-// resolvePublicGitHub returns a GitHub instance for the public GitHub API if
-// the environment is a GitHub Enterprise instance.
-function resolvePublicGitHub(
-  github: InstanceType<typeof GitHub>,
-  inputs: Inputs,
-): InstanceType<typeof GitHub> {
-  const apiUrl = process.env.GITHUB_API_URL || ``;
-  if (apiUrl.startsWith(publicGitHubApiUrl)) {
-    core.info("Running on GitHub.com, using default GitHub API.");
-    return github;
-  }
-  const publicToken = inputs.public_github_token;
-  if (!publicToken) {
-    throw new Error("public_github_token is required for public GitHub API");
-  }
-  const publicGitHub = getOctokit(publicToken, { baseUrl: publicGitHubApiUrl });
-  core.info("Running on GitHub Enterprise, using public GitHub API.");
-  return publicGitHub;
 }
 
 // resolveVersion from the input or environment.
@@ -143,7 +120,7 @@ type platformTable = {
 };
 
 // downloadBuf downloads the buf binary and returns the path to the binary.
-async function downloadBuf(version: string): Promise<string> {
+async function downloadBuf(version: string, github_token: string): Promise<string> {
   const table: platformTable = {
     darwin: {
       x64: "buf-Darwin-x86_64",
@@ -172,7 +149,7 @@ async function downloadBuf(version: string): Promise<string> {
   }
   const downloadURL = `https://github.com/bufbuild/buf/releases/download/v${version}/${executable}`;
   try {
-    return await tc.downloadTool(downloadURL);
+    return await tc.downloadTool(downloadURL, undefined, github_token);
   } catch (error) {
     throw new Error(
       `Failed to download buf version ${version} from "${downloadURL}": ${error}`,

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -119,7 +119,10 @@ type platformTable = {
 };
 
 // downloadBuf downloads the buf binary and returns the path to the binary.
-async function downloadBuf(version: string, githubToken: string): Promise<string> {
+async function downloadBuf(
+  version: string,
+  githubToken: string,
+): Promise<string> {
   const table: platformTable = {
     darwin: {
       x64: "buf-Darwin-x86_64",

--- a/src/main.ts
+++ b/src/main.ts
@@ -37,19 +37,19 @@ const publicGitHubApiUrl = "https://api.github.com";
 async function main() {
   const inputs = getInputs();
   const github = getOctokit(inputs.github_token);
-  let publicGitHubToken = inputs.github_token;
-  let publicGitHub = github;
-  const apiUrl = github.request.defaults.baseUrl || process.env.GITHUB_API_URL || ``;
+  let publicGithubToken = inputs.github_token;
+  let publicGithub = github;
+  const apiUrl = process.env.GITHUB_API_URL || ``;
   if (!apiUrl.startsWith(publicGitHubApiUrl)) {
     core.info("Running on GitHub Enterprise, using public GitHub API.");
-    publicGitHubToken = inputs.public_github_token;
-    if (publicGitHubToken == "") {
+    publicGithubToken = inputs.public_github_token;
+    if (publicGithubToken == "") {
       // Warn if the public GitHub token is not set. Don't fail as not required.
       core.warning("public_github_token not set, GitHub API requests may be limited");
     }
-    publicGitHub = getOctokit(inputs.public_github_token, { baseUrl: publicGitHubApiUrl });
+    publicGithub = getOctokit(publicGithubToken, { baseUrl: publicGitHubApiUrl });
   }
-  const [bufPath, bufVersion] = await installBuf(publicGitHub, publicGitHubToken, inputs.version);
+  const [bufPath, bufVersion] = await installBuf(publicGithub, publicGithubToken, inputs.version);
   core.setOutput(Outputs.BufVersion, bufVersion);
   core.setOutput(Outputs.BufPath, bufPath);
   core.saveState(Outputs.BufPath, bufPath);

--- a/src/main.ts
+++ b/src/main.ts
@@ -30,11 +30,26 @@ import { installBuf } from "./installer";
 import { findCommentOnPR, commentOnPR } from "./comment";
 import { parseModuleNames, ModuleName } from "./config";
 
+// URL for the public GitHub API.
+const publicGitHubApiUrl = "https://api.github.com";
+
 // main is the entrypoint for the action.
 async function main() {
   const inputs = getInputs();
   const github = getOctokit(inputs.github_token);
-  const [bufPath, bufVersion] = await installBuf(github, inputs);
+  let publicGitHubToken = inputs.github_token;
+  let publicGitHub = github;
+  const apiUrl = github.request.defaults.baseUrl || process.env.GITHUB_API_URL || ``;
+  if (!apiUrl.startsWith(publicGitHubApiUrl)) {
+    core.info("Running on GitHub Enterprise, using public GitHub API.");
+    publicGitHubToken = inputs.public_github_token;
+    if (publicGitHubToken == "") {
+      // Warn if the public GitHub token is not set. Don't fail as not required.
+      core.warning("public_github_token not set, GitHub API requests may be limited");
+    }
+    publicGitHub = getOctokit(inputs.public_github_token, { baseUrl: publicGitHubApiUrl });
+  }
+  const [bufPath, bufVersion] = await installBuf(publicGitHub, publicGitHubToken, inputs.version);
   core.setOutput(Outputs.BufVersion, bufVersion);
   core.setOutput(Outputs.BufPath, bufPath);
   core.saveState(Outputs.BufPath, bufPath);

--- a/src/main.ts
+++ b/src/main.ts
@@ -37,6 +37,8 @@ const publicGitHubApiUrl = "https://api.github.com";
 async function main() {
   const inputs = getInputs();
   const github = getOctokit(inputs.github_token);
+  // Check if the action is running on a GitHub Enterprise instance.
+  // If so, use the public GitHub API for resolving the Buf version etc.
   let publicGithubToken = inputs.github_token;
   let publicGithub = github;
   const apiUrl = process.env.GITHUB_API_URL || ``;
@@ -45,11 +47,19 @@ async function main() {
     publicGithubToken = inputs.public_github_token;
     if (publicGithubToken == "") {
       // Warn if the public GitHub token is not set. Don't fail as not required.
-      core.warning("public_github_token not set, GitHub API requests may be limited");
+      core.warning(
+        "public_github_token not set, GitHub API requests may be limited",
+      );
     }
-    publicGithub = getOctokit(publicGithubToken, { baseUrl: publicGitHubApiUrl });
+    publicGithub = getOctokit(publicGithubToken, {
+      baseUrl: publicGitHubApiUrl,
+    });
   }
-  const [bufPath, bufVersion] = await installBuf(publicGithub, publicGithubToken, inputs.version);
+  const [bufPath, bufVersion] = await installBuf(
+    publicGithub,
+    publicGithubToken,
+    inputs.version,
+  );
   core.setOutput(Outputs.BufVersion, bufVersion);
   core.setOutput(Outputs.BufPath, bufPath);
   core.saveState(Outputs.BufPath, bufPath);

--- a/src/main.ts
+++ b/src/main.ts
@@ -34,7 +34,7 @@ import { parseModuleNames, ModuleName } from "./config";
 async function main() {
   const inputs = getInputs();
   const github = getOctokit(inputs.github_token);
-  const [bufPath, bufVersion] = await installBuf(github, inputs.version);
+  const [bufPath, bufVersion] = await installBuf(github, inputs);
   core.setOutput(Outputs.BufVersion, bufVersion);
   core.setOutput(Outputs.BufPath, bufPath);
   core.saveState(Outputs.BufPath, bufPath);

